lib/resqued/master_state.rb

```diff
@@ -0,0 +1,71 @@
+module Resqued
+  class MasterState
+    def initialize
+      @listeners_created = 0
+      @listener_states = {}
+    end
+
+    # Public: When starting fresh, from command-line options, assign the initial values.
+    def init(options)
+      @config_paths = options.fetch(:config_paths)
+      @exec_on_hup  = options.fetch(:exec_on_hup) { false }
```

Comment:
Since the block here is simple and the default value too, I think this can be `options.fetch(:exec_on_hup, false)` for all these here?